src/signer/interface.ts

```diff
@@ -38,6 +84,38 @@ export abstract class SignerInterface {
    *  - maxFee<br/>
    *  - version<br/>
    *  - nonce<br/>
```

Comment:
I don't have the patience to verify until the end of the file.
Please apply the previous comments in the remaining code, and more than all : TEST YOUR EXAMPLES in a node script.